### PR TITLE
integrating bounded l2 norm fixed point vector sum types.

### PR DESCRIPTION
**Pull request by ooovi**
As discussed [here](https://github.com/divviup/janus/issues/464), this PR enables using our [prio3 type for aggregating vectors of fixed-point numbers](https://github.com/divviup/libprio-rs/pull/283) with janus. It mainly adds the enum cases and match branches required to dispatch on the 16-, 32- and 64-bit variants of fixed-point vector aggregation.

One notable thing I did is adding a [helper enum to enable tagging the JSON from the upload request with the fixedpoint type in the end-to-end interop test](https://github.com/dpsa-project/janus/blob/abda6a06637014808cdfead37912d88b2ca972bb/interop_binaries/src/bin/janus_interop_client.rs#L30-L43) only for our vectors, as the standard string representation of fixedpoint numbers does not carry that information.

Looking forward to your comments :)

### CODE REVIEW

**Review by tgeoghegan — COMMENTED** [review]
This change looks sound to me, but the Janus team is debating how we want to handle this. For one thing, we may want to gate this VDAF behind a feature (much as the fixed point implementation in `libprio-rs` is gated behind feature `experimental`).

File: collector/src/lib.rs [hunk 1]
```diff
@@ -822,7 +823,7 @@ mod tests {
         let collector = setup_collector(vdaf);
 
         let batch_interval = Interval::new(
-            Time::from_seconds_since_epoch(1_000_000),
+            Time::from_seconds_since_epoch(2_000_000), // TODO why???
```
Comment (branlwyd) [inline collector/src/lib.rs:826]: Not sure offhand -- if you don't know why this change is necessary, I think it's OK to just drop the TODO.

**Review by branlwyd — CHANGES_REQUESTED** [review]
As Tim mentioned, let's gate this with crate features, disabled by default. (Tests can enable the features in order to allow testing this functionality by default.)

File: collector/src/lib.rs (hunk 1 above)
Comment (divergentdave) [reply collector/src/lib.rs:826]: I saw a test flake in this module yesterday, so it was probably random chance, unconnected to this timestamp. (I suspect we may be running into a `backoff` timer we need to tweak, but I haven't had a chance to dig into it yet)

**Comment by ooovi** [comment]
we are now hiding behind a feature.

> Tests can enable the features in order to allow testing this functionality by default
we had to adjust the `Dockerfile.interop` and `Dockerfile.interop_aggregator` to enable the feature during tests. is that what you meant, or should we do something more?

**Review by branlwyd — COMMENTED** [review]
> we had to adjust the Dockerfile.interop and Dockerfile.interop_aggregator to enable the feature during tests. is that what you meant, or should we do something more?

I meant to modify the dependency in the `dev-dependencies` so that it enables the new feature, allowing the relevant tests to _not_ be gated by the feature & thus tested by `cargo test` without additional flags. (My goal is to make a release binary not include these "experimental"/non-standardized VDAFs by default, but I think it is wise to test them by default.)

See https://github.com/divviup/janus/blob/f8ad51953c0a9d31e6eaf973c882c62287844caf/aggregator/Cargo.toml#L98 for an example of the change to `Cargo.toml` -- there are a number of `dev-dependencies` throughout the workspace that enable the `test-util` feature.

**Comment by ooovi** [comment]
> I meant to modify the dependency in the `dev-dependencies` so that it enables the new feature, allowing the relevant tests to _not_ be gated by the feature & thus tested by `cargo test` without additional flags. (My goal is to make a release binary not include these "experimental"/non-standardized VDAFs by default, but I think it is wise to test them by default.)

I did this (and the other changes you suggested) in the latest commit. Is there another way of enabling the feature on the containers used in the interop end-to-end test except passing the feature flag to the build command in the `Dockerfile.interop` and `Dockerfile.interop_aggregator`?

File: collector/src/lib.rs [hunk 2]
```diff
@@ -822,7 +824,7 @@ mod tests {
         let collector = setup_collector(vdaf);
 
         let batch_interval = Interval::new(
-            Time::from_seconds_since_epoch(1_000_000),
+            Time::from_seconds_since_epoch(2_000_000),
```
Comment (divergentdave) [inline collector/src/lib.rs:827]: This change can be backed out, #916 properly fixes this test.

**Review by branlwyd — APPROVED** [review]

**Comment by ooovi** [comment]
I merged the current master, as there were conflicting changes. Also squashed into one commit. Could you run the workflow again? :)

**Comment by ooovi** [comment]
the workflow cannot be run on forks because they have no access to your github secrets. there seems to be an event that gives them access:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
but i don't know if that's what you want to be doing in general:
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Comment by branlwyd** [comment]
I ran the tests locally; I am OK merging with that level of testing. I would prefer not to enable running with secrets against forks; perhaps if you will be creating many PRs, we can flip the appropriate bits so that you can create a branch on the main repository, and then we can enable you to run CI directly.
